File: tests/sdk-examples/go/tdf.go
// Package examples contains compile-only type-assertion tests for the OpenTDF Go SDK.
// These functions are never run; they exist solely so that `go build ./...` will fail
// if a documented method signature drifts from the actual SDK.
package examples

import (
	"bytes"
	"context"
	"io"

	"github.com/opentdf/platform/lib/ocrypto"
	"github.com/opentdf/platform/sdk"
)

// verifyCreateTDF verifies: (s SDK) CreateTDF(io.Writer, io.ReadSeeker, ...TDFOption) (*TDFObject, error)
func verifyCreateTDF(client sdk.SDK, w io.Writer, r io.ReadSeeker) {
	var obj *sdk.TDFObject
	var err error
	obj, err = client.CreateTDF(w, r)
	_, _ = obj, err
}

// verifyLoadTDF verifies: (s SDK) LoadTDF(io.ReadSeeker, ...TDFReaderOption) (*Reader, error)
func verifyLoadTDF(client sdk.SDK, rs io.ReadSeeker) {
	var r *sdk.Reader
	var err error
	r, err = client.LoadTDF(rs)
	_, _ = r, err
}

// verifyIsValidTdf verifies: IsValidTdf(io.ReadSeeker) (bool, error)
func verifyIsValidTdf(rs io.ReadSeeker) {
	var ok bool
	var err error
	ok, err = sdk.IsValidTdf(rs)
	_, _ = ok, err
}

// verifyBulkDecrypt verifies: (s SDK) BulkDecrypt(ctx, ...BulkDecryptOption) error
func verifyBulkDecrypt(client sdk.SDK, ctx context.Context) {
	var err error
	err = client.BulkDecrypt(ctx)
	_ = err
}

// verifyPrepareBulkDecrypt verifies: (s SDK) PrepareBulkDecrypt(ctx, ...) (*BulkDecryptPrepared, error)
func verifyPrepareBulkDecrypt(client sdk.SDK, ctx context.Context) {
	var p *sdk.BulkDecryptPrepared
	var err error
	p, err = client.PrepareBulkDecrypt(ctx)
	_, _ = p, err
}

// verifyReaderMethods verifies Reader method signatures
func verifyReaderMethods(r *sdk.Reader) {
	var w bytes.Buffer

	var n int64
	var err error
	n, err = r.WriteTo(&w)
	_, _ = n, err

	var meta []byte
	meta, err = r.UnencryptedMetadata()
	_, _ = meta, err

	var pol sdk.PolicyObject
	pol, err = r.Policy()
	_, _ = pol, err

	var attrs []string
	attrs, err = r.DataAttributes()
	_, _ = attrs, err

	var m sdk.Manifest
	m = r.Manifest()
	_ = m
}

// verifyEncryptOptions verifies encrypt option function signatures
func verifyEncryptOptions() {
	_ = sdk.WithDataAttributes("https://example.com/attr/clearance/value/executive")
	_ = sdk.WithKasInformation(sdk.KASInfo{URL: "https://kas.example.com"})
	_ = sdk.WithAutoconfigure(true)
	_ = sdk.WithMetaData("metadata string")
	_ = sdk.WithMimeType("text/plain")
	_ = sdk.WithSegmentSize(int64(1024 * 1024))
	_ = sdk.WithAssertions(sdk.AssertionConfig{})
	_ = sdk.WithWrappingKeyAlg(ocrypto.EC256Key)
}

// verifyDecryptOptions verifies decrypt option function signatures
func verifyDecryptOptions() {
	_ = sdk.WithKasAllowlist([]string{"https://kas.example.com"})
	_ = sdk.WithIgnoreAllowlist(true)
	_ = sdk.WithAssertionVerificationKeys(sdk.AssertionVerificationKeys{})
	_ = sdk.WithDisableAssertionVerification(true)
	_ = sdk.WithSessionKeyType(ocrypto.EC256Key)
	_ = sdk.WithTDFFulfillableObligationFQNs([]string{"https://example.com/obl/audit/value/log"})
	_ = sdk.WithMaxManifestSize(int64(1024 * 1024))
}

File: tests/sdk-examples/ts/tdf.ts
/**
 * Compile-only type-assertion tests for the OpenTDF TypeScript SDK.
 * These functions are never run; they exist solely so that `tsc --noEmit` will fail
 * if a documented method signature or option property drifts from the actual SDK.
 *
 * Return-type annotations act as the type assertions:
 *   async function f(): Promise<DecoratedStream>  →  compile error if createZTDF return type changes
 *   function f(): TDFReader                        →  compile error if open return type changes
 */

import type {
  CreateZTDFOptions,
  DecoratedStream,
  ReadOptions,
  Source,
  TDFReader,
} from "@opentdf/sdk";
import { OpenTDF } from "@opentdf/sdk";

// A minimal Source value used to populate required option fields without runtime deps.
const mockSource: Source = { type: "buffer", location: new Uint8Array() };

/**
 * Verifies: createZTDF(opts: CreateZTDFOptions) => Promise<DecoratedStream>
 * Also verifies key CreateZTDFOptions property names and types.
 */
async function verifyCreateZTDF(client: OpenTDF): Promise<DecoratedStream> {
  const opts: CreateZTDFOptions = {
    source: mockSource,
    attributes: ["https://example.com/attr/clearance/value/executive"],
    defaultKASEndpoint: "https://kas.example.com",
    mimeType: "text/plain",
    metadata: { info: "example" },
    splitPlan: [{ kas: "https://kas.example.com" }],
    autoconfigure: true,
  };
  return client.createZTDF(opts);
}

/**
 * Verifies: read(opts: ReadOptions) => Promise<DecoratedStream>
 * Also verifies key ReadOptions property names and types.
 */
async function verifyRead(client: OpenTDF): Promise<DecoratedStream> {
  const opts: ReadOptions = {
    source: mockSource,
    platformUrl: "https://platform.example.com",
    allowedKASEndpoints: ["https://kas.example.com"],
    ignoreAllowlist: false,
    noVerify: false,
  };
  return client.read(opts);
}

/**
 * Verifies: open(opts: ReadOptions) => TDFReader  (synchronous, no Promise)
 */
function verifyOpen(client: OpenTDF): TDFReader {
  const opts: ReadOptions = { source: mockSource };
  return client.open(opts);
}

// Suppress unused-variable warnings so the file compiles under --strict
export { verifyCreateZTDF, verifyOpen, verifyRead };
